tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/arena/ElemPtr.scala

```diff
@@ -20,21 +20,23 @@ sealed trait ElemPtr {
    * Translate the membership test into an expression that can be understood by Z3SolverContext.
    */
   def toSmt: TlaEx
+
+  /**
+   * After certain set operations, every pointer must become a SmtExprElemPtr, because the operation invalidates the
+   * guarantees of e.g. FixedElemPtr.
+   */
+  def generalize: SmtExprElemPtr = SmtExprElemPtr(elem, toSmt)
```

Comment:
Do we still need this one?

Reply:
Yes, you'll want this when you e.g. call filter. Filtering a fixed pointer voids the membership guarantee.